jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java

```diff
@@ -31,53 +32,73 @@
  */
 public class ImageReference {
 
+  /** Regular expression patterns for the image reference. */
+  private static class Regex extends RegexBuilder {
+
+    /**
+     * Matches all sequences of alphanumeric characters possibly separated by any number of dashes
+     * in the middle.
+     */
+    private static final String REGISTRY_PART =
+        group(alphanum(), group(any(alphanum('-')), alphanum()));
+
+    /**
+     * Matches sequences of {@link #REGISTRY_PART} separated by a dot, with an optional {@code
+     * :port} at the end.
+     */
+    private static final String REGISTRY =
```

Comment:
I'm a little skeptical of this change.  It seems to replace something that I'm very familiar with (regex strings),  with a sort of custom DSL that I'd have to do a bunch of work to understand.

Reply:
Yeah, it does provide a custom DSL that requires some extra work to understand. The intention was to try to follow what Docker does for its reference regex: https://github.com/docker/distribution/blob/master/reference/regexp.go